src/cartographer/config/overlays/image-pull-secret.yaml

```diff
@@ -0,0 +1,26 @@
+# Copyright 2022 VMware
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#@ load("@ytt:overlay", "overlay")
+
+#! TODO(scothis) align how cartographer and cartographer-conventions manage pull secrets
```

Comment:
oh that's a nice overlay - going w/ this, we could get rid of that hardcoded name we have there right now for the placeholder secret, right? (https://github.com/vmware-tanzu/cartographer/blob/feb98ff36248e68ed0df428a2f885d599a4db1ad/config/manager/deployment.yaml#L54-L55) 

I guess we could _also_ get rid of the placeholder secret itself foo (https://github.com/vmware-tanzu/cartographer/blob/feb98ff36248e68ed0df428a2f885d599a4db1ad/config/manager/deployment.yaml#L23-L32) and have it managed here in `package-for-cartographer`?

Reply:
done. This PR now defines a new secret and appends it to the imagePullSecrets for all Deployments in the cartographer-system namespace